google/gax/__init__.py

```diff
@@ -464,10 +468,14 @@ def __iter__(self):
         return self
 
     def next(self):
+        """For Python 2.7 compatibility; see __next__."""
+        return self.__next__()
+
+    def __next__(self):
         """Retrieves the next resource."""
         # pylint: disable=next-method-called
         while not self._current:
-            self._current = self._iterator.next()
+            self._current = next(self._iterator)
```

Comment:
I see. Is this a canonical pattern for Python2/3 compatible code, or should it be explained more in the iterator `next()` docstrings?